CMakeLists.txt

```diff
@@ -59,6 +59,27 @@ endif()
 # Add cmake modules directory to the path.
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
 
+set(CA_CMAKE_VERBOSE_LEVEL ""  CACHE STRING "cmake verbosity level 2 detailed (default), 1 summary, 0 none")
+set(OCK_CMAKE_VERBOSE_LEVEL "2")
+
+# Check if we are in an LLVM Tree.
+# This is done by checking for the LLVMCore target which does not exist
+# until we import LLVM.
+# This only happens of we have not imported or are fetched by LLVM via
+# FetchContent or as an external project. In this case we set a cmake
+# variable OCK_IN_LLVM_TREE to TRUE.
+set(OCK_IN_LLVM_TREE FALSE)
+if (TARGET LLVMCore)
+  set(OCK_IN_LLVM_TREE TRUE)
+  # We always reduce verbosity level if included in LLVM tree
+  set(OCK_CMAKE_VERBOSE_LEVEL "0")
+endif()
```

Comment:
We could just keep the default as always verbose, and override it from within DPC++.

Reply:
I did think about that, I was worried that we end up adding loads and loads of cmake options before including it and maybe this was another way.

Reply:
If DPC++ native CPU wants loads of things different from what we want as the default, there's probably a bigger problem that we should discuss. I would expect that a list of just those things where DPC++ wants something different to be relatively short, shorter than repeated `OCK_IN_LLVM_TREE` checks. Is that not the case?